packages/api/src/router/system.ts

```
@@ -118,6 +118,25 @@ export const systemRouter = createTRPCRouter({
+        canUser.perform(Permission.SystemGet).on({ type: "system", id: input }),
+    })
+    .input(z.string().uuid())
+    .query(({ ctx: { db }, input }) =>
+      db
+        .select()
+        .from(system)
+        .innerJoin(workspace, eq(system.workspaceId, workspace.id))
+        .leftJoin(environment, eq(environment.systemId, system.id))
+        .where(eq(system.id, input))
+        .then((rows) => ({
+          ...rows[0]!.system,
```

Biome

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)